civihr_employee_portal/civihr_employee_portal.module

```diff
@@ -854,12 +854,100 @@ function civihr_employee_portal_views_default_views() {
 
     if (isset($view)) {
       $views[$view->name] = $view;
+      civihr_employee_portal_fix_custom_fields_mapping($view);
     }
   }
 
   return $views;
 }
 
+/**
+ * To fix mapping on Views import of CiviCRM custom fields
+ *
+ * @param view $view
+ *     the view that is being imported / recreated
+ */
+function civihr_employee_portal_fix_custom_fields_mapping($view) {
+  global $databases;
+  // iterating displays in view
+  foreach ($view->display as $display) {
+    // iterating options on display: fields, filters, arguments
+    foreach ($display->display_options as &$display_option) {
+      if (!is_array($display_option)) {
+        continue;
+      }
+      foreach ($display_option as &$option) {
+        // only continue if this option is about a field
+        if (!isset($option['field'])) {
+          continue;
+        }
+        $option_id_parts = explode('_', $option['field']);
+        $final_id_part = array_pop($option_id_parts);
+        // it must contain underscores and a number at the end to be checked
+        if (!$option_id_parts > 0 || !intval($final_id_part)) {
+          continue;
+        }
+        // checking if this table is part of the civicrm views integration
+        if ( isset($databases['default']['default']['prefix'][$option['table']]) ) {
+          $prefix_table = $databases['default']['default']['prefix'][$option['table']];
+          $table = $prefix_table.$option['table'];
```

Comment:
The table names are also prefixed with ids. Did you try do check if these IDs can also differ from site to site? I think they can, depending on how the custom groups are created

Reply:
@davialexandre  Good point. Tables are now checked too with a similar mechanism as fields, so while tables keep their base name no matter if their number at the end differs, the import will be fine. I tested it manually changing table names on an exported view.